chirp/drivers/tdh8.py

```diff
@@ -0,0 +1,2153 @@
+                mem.freq = ham_freqs
+                _mem.rxfreq = _mem.txfreq = mem.freq / 10
+                mem.power = TX_POWER[1]
+                mem.tmode = ""
+
+                # number 189 - 199 can not be edited
+                mem.immutable = [
+                    "freq", "name", "tmode", "power",
+                    "duplex", "freqhop", "pttid", "bcl"]
+
+            else:
+                _mem.rxfreq = mem.freq / 10
+            # elif (mem.number >= 1 and mem.number <= 188
+            #       ) and (
+            #           (mem.freq >= 144000000 and mem.freq <= 148000000) or (
+            #               mem.freq >= 420000000 and mem.freq <= 450000000)):
+            #     _mem.rxfreq = mem.freq / 10
+            # else:
+            #     msg = (
+            #         "The frequency in channels 1-188 must be "
+            #         "between 144.00000-148.00000 or 420.00000-450.00000")
+            #     raise InvalidValueError(msg)
+
+        elif self._gmrs:
+            if (mem.number >= 189 and mem.number <= 199):
+                ham_freqs = int(HAM_GMRS_FREQS[mem.number - 189] * 1000000)
+                mem.freq = ham_freqs
+                _mem.rxfreq = mem.freq / 10
+                mem.duplex = ""
+                mem.power = TX_POWER[1]
+                mem.tmode = ""
+
+                # number 189 - 199 can not be edited
+                mem.immutable = [
+                    "freq", "name", "tmode", "power",
+                    "duplex", "freqhop", "pttid", "bcl"]
+
+            elif (mem.number >= 1 and mem.number <= 30):
+                if mem.number >= 8 and mem.number <= 14:
+                    gmrs_freqs = int(GMRS_FREQS_B[mem.number - 1] * 1000000)
+                    mem.freq = gmrs_freqs
+                    _mem.rxfreq = mem.freq / 10
+                    mem.power = TX_POWER[0]
+                    mem.mode = "NFM"
+                    mem.duplex = ""
+                else:
+                    gmrs_freqs = int(GMRS_FREQS_B[mem.number - 1] * 1000000)
+                    mem.freq = gmrs_freqs
+                    mem.duplex = ""
+                    _mem.rxfreq = mem.freq / 10
+
+            elif float("%i.%06i" % (mem.freq / 1000000, mem.freq % 1000000)
+                       ) in GMRS_FREQ_A:
+                _mem.rxfreq = mem.freq / 10
+                _mem.offset = 2
+                mem.offset = 5
+                _mem.txfreq = (mem.freq + 5000000) / 10
+
+            elif (mem.number >= 55 and mem.number <= 188) and (
+                (mem.freq >= 136000000 and mem.freq <= 174000000) or (
+                    mem.freq >= 400000000 and mem.freq <= 520000000)):
+                _mem.rxfreq = mem.freq / 10
+
+            else:
+                _mem.rxfreq = mem.freq / 10
+                # msg = (
+                #     "The frequency in channels 31-55 must be between "
+                #     "462.55000-462.72500 in 0.025 increments.")
+                # raise errors.RadioError(msg)
+
```

Comment:
I have some questions and don’t know how to solve them. In gmrs mode, channel 31-55 is a fixed optional channel, fixed at 462.55000-462.725 with an increment of 0.025, which is:
```
462.550
462.575
432.600
462.650
462.675
462.700
462.725
```
If I use error to prompt the user here, will it pass the tox-driver test.
what should i do here？